.github/workflows/main.yml

```diff
@@ -27,82 +27,159 @@ jobs:
     steps:
     - name: Checkout Code
       uses: actions/checkout@v2
-
-    - name: Checkout Submodules
-      run: git submodule update --init --recursive
+      with:
+        submodules: 'recursive'
 
     - name: Install Python
       uses: actions/setup-python@v2
       with:
         python-version: '3.x'
 
-    - name: Install Build Tools (pip)
-      run: pip install meson ninja zstandard
-
     - name: Install Build Tools (apt)
-      run: sudo apt update && sudo apt install build-essential libsdl2-dev libogg-dev libopusfile-dev libpng-dev libzip-dev libx11-dev libwayland-dev python3-docutils libwebp-dev libfreetype6-dev libzstd-dev
-
-    - name: Configure - Step 1 (Meson)
-      run: meson setup --prefix=$(pwd)/build-test --wrap-mode=nofallback --force-fallback-for=cglm build/ -Db_lto=false -Dstrip=false -Ddeprecation_warnings=no-error -Dwerror=true -Dbasis_universal:werror=false -Db_pch=false
-
-    - name: Build - Step 2 (Ninja)
-      run: meson compile -C build/
-
-    - name: Install - Step 3 (Meson)
-      run: meson install -C build/
-
-    - name: Run - Step 4 (Taisei)
+      run: |
+        sudo apt update
+        sudo apt install -y -qq \
+        build-essential \
+        libsdl2-dev \
+        libogg-dev \
+        libopusfile-dev \
+        libpng-dev \
+        libzip-dev \
+        libx11-dev \
+        libwayland-dev \
+        python3-docutils \
+        libwebp-dev \
+        libfreetype6-dev \
+        libzstd-dev
+        # pip
+        pip install \
+        meson==0.55.3 \
+        ninja \
+        zstandard
+
+    # TODO: turn -Dwerror back on when build is fixed
+    - name: Configure (Meson)
+      run: |
+        meson setup build/ \
+          --prefix=$(pwd)/build-test \
+          --native-file scripts/ci/linux-x86_64-build-test-ci.ini
```

Comment:
I'm not sure `scripts/` is the right place for this. How about putting both the machine files and the test replay into `misc/ci`, along with any further CI-only files we might need in the future?